card/src/main/java/com/adyen/checkout/card/ui/AddressFormInput.kt

```diff
@@ -110,19 +114,27 @@ class AddressFormInput @JvmOverloads constructor(
             setAdapter(countryAdapter)
             onItemClickListener = AdapterView.OnItemClickListener { _, _, position, _ ->
                 val selectedCountryCode = countryAdapter.getItem(position).code
-                if (delegate.outputData.addressState.country.value != selectedCountryCode) {
-                    delegate.updateInputData {
-                        address.reset()
-                        address.country = selectedCountryCode
+                if (delegate.addressOutputData.country.value != selectedCountryCode) {
+                    delegate.updateAddressInputData {
+                        reset()
+                        country = selectedCountryCode
                     }
                     populateFormFields(AddressSpecification.fromString(selectedCountryCode))
                 }
             }
         }
     }
 
-    internal fun attachDelegate(delegate: CardDelegate) {
+    internal fun attachDelegate(delegate: AddressDelegate, coroutineScope: CoroutineScope) {
         this.delegate = delegate
+        subscribeCountryAndStateList(coroutineScope)
+    }
+
+    private fun subscribeCountryAndStateList(coroutineScope: CoroutineScope) {
+        delegate.addressOutputDataFlow.distinctUntilChanged().onEach { addressOutputData ->
```

Comment:
Doesn't `StateFlow`s have `distinctUntilChanged()` applied by default? `distinctUntilChanged()` might be unnecessary here